configuration/interface.go

```diff
@@ -38,7 +41,7 @@ type Client interface {
 	// WatchForChanges sets up a Consul watch for the target key and send back updates on the update channel.
 	// Passed in struct is only a reference for Configuration service, empty struct is ok
 	// Sends the configuration in the target struct as interface{} on updateChannel, which caller must cast
-	WatchForChanges(updateChannel chan<- interface{}, errorChannel chan<- error, configuration interface{}, waitKey string)
+	WatchForChanges(updateChannel chan<- interface{}, errorChannel chan<- error, configuration interface{}, waitKey string, msgClient messaging.MessageClient)
```

Comment:
Can't pass `messaging.MessageClient` since it hasn't been created since the configuration needed to create it is in the Config Provider. This API gets called from go-mod-bootstrap before the the `MessageClient` is created.

https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/config/config.go#L784
https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/config/config.go#L846 https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/config/config.go#L661

IMO, the better way to approach this is to add another API called `SetMessageClient` to the abstract API. The Consul implementation simply returns ignoring the client. The keep implementation saves the reference where `WatchForChanges` can use it. `WatchForChanges` will have to pause until `SetMessageClient` is called.

Then go-mo-bootstrap can call `SetMessageClient` once the `MessageClient`  has been created.

Reply:
We have designed similar mechanism [here](https://github.com/jackchenjc/go-mod-bootstrap/commit/057ab5ec3004d7a2fd8b042616bb3efd84a9226f#diff-5346c73553fe1e8350deb1a29ed066b1202c1ea35e57d206adbb6f359ed1d299R810-R825) in go-mod-bootstrap, may I know which option you believe would be preferable?

Reply:
OK, I that makes sense now.